packages/@cdktf/hcl2cdk/lib/index.ts

```diff
@@ -0,0 +1,470 @@
+    if (item && "dynamic" in item) {
+      const dyn = (item as any)["dynamic"];
+      const { for_each, ...others } = dyn;
+      const dynamicRef = Object.keys(others)[0];
+      return [
+        ...references,
+        ...findUsedReferences([...nodeIds, dynamicRef], dyn),
+      ];
+    }
+    return [
+      ...references,
+      ...Object.values(item as Record<string, any>).reduce(
+        (carry, i) => [...carry, ...findUsedReferences(nodeIds, i)],
+        []
+      ),
+    ];
+  }
+
+  if (typeof item === "string") {
+    const extractedRefs = extractReferencesFromExpression(item, nodeIds);
+    return [...references, ...extractedRefs];
+  }
+  return references;
+}
+
+function asExpression(
+  type: string,
+  name: string,
+  config: any,
+  nodeIds: string[],
+  reference?: Reference
+) {
+  const isNamespacedImport = type.includes(".");
+  const subject = isNamespacedImport
+    ? t.memberExpression(
+        t.identifier(type.split(".")[0]), // e.g. aws
+        t.identifier(pascalCase(type.split(".")[1])) // e.g. NatGateway
+      )
+    : t.identifier(pascalCase(type));
+
+  const expression = t.newExpression(subject, [
+    t.thisExpression(),
+    t.stringLiteral(name),
+    valueToTs(config, nodeIds),
+  ]);
+  return reference
+    ? t.variableDeclaration("const", [
+        t.variableDeclarator(
+          t.identifier(referenceToVariableName(reference)),
+          expression
+        ),
+      ])
+    : t.expressionStatement(expression);
+}
+function output(
+  key: string,
+  _id: string,
+  item: Output,
+  graph: DirectedGraph
+): t.Statement {
+  const nodeIds = graph.nodes();
+  const [{ value, description, sensitive }] = item;
+
+  return asExpression(
+    "TerraformOutput",
+    key,
+    {
+      value,
+      description,
+      sensitive,
+    },
+    nodeIds
+  );
+}
+
+function variable(
+  key: string,
+  id: string,
+  item: Variable,
+  graph: DirectedGraph
+): t.Statement {
+  // We don't handle type information right now
+  const [{ type, ...props }] = item;
+  const nodeIds = graph.nodes();
+
+  return asExpression(
+    "TerraformVariable",
+    key,
+    props,
+    nodeIds,
+    getReference(graph, id)
+  );
+}
+
+function local(
+  key: string,
+  _id: string,
+  item: any,
+  graph: DirectedGraph
+): t.Statement {
+  const nodeIds = graph.nodes();
+  return t.variableDeclaration("const", [
+    t.variableDeclarator(
+      t.identifier(camelCase(key)),
+      valueToTs(item, nodeIds)
+    ),
+  ]);
+}
+
+function modules(
+  key: string,
+  id: string,
+  item: Module,
+  graph: DirectedGraph
+): t.Statement {
+  const [{ source, ...props }] = item;
+  const nodeIds = graph.nodes();
+
+  return asExpression(source, key, props, nodeIds, getReference(graph, id));
+}
+
+function provider(
+  key: string,
+  _id: string,
+  item: Provider,
+  graph: DirectedGraph
+): t.Statement {
+  const nodeIds = graph.nodes();
+  const props = item[0];
+
+  if (props.alias) {
+    throw new Error(
```

Comment:
Would this be hard to add?

Reply:
No, I just wanted to be explicit about what is currently supported :)